test/task.cc

```diff
@@ -534,3 +580,23 @@ TEST(Task, RefSuccess) {
 
   EXPECT_EQ(110, x);
 }
+
+TEST(Task, RaisesWith) {
+  auto e = []() {
+    return Task::Of<int>::Raises<std::runtime_error>::With<int, std::string>(
+        42,
+        "hello world",
+        [](auto i, auto s) {
+          EXPECT_EQ(s, "hello world");
+          return Just(i)
+              .template raises<std::runtime_error>();
```

Comment:
That's some kung fu magic! Let's be explicit for our poor readers, and then if the `Just()` implementation ever changes in the future this line of code won't break awkwardly.
```suggestion
return Eventual<int>()
    .template raises<std::runtime_error>()
    .start([&](auto& k) {
      return k.Start(i);
    });
```